internal/repository/account_repo_test.go

```diff
@@ -0,0 +1,251 @@
+package repository
+
+import (
+	"context"
+	"testing"
+
+	"github.com/gopay/internal/models"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestAccount_Create(t *testing.T) {
+	id := "0003"
+	idGenerator := func() string {
+		return id
+	}
+
+	type args struct {
+		ctx      context.Context
+		name     string
+		lastname string
+		data     map[string]models.Account
+	}
+
+	var scenarios = map[string]struct {
+		given   args
+		want    models.Account
+		wantErr error
+	}{
+		"happy-path": {
+			given: args{
+				ctx:      context.Background(),
+				name:     "Caio",
+				lastname: "Henrique",
+				data: map[string]models.Account{
+					"0001": {
+						AccountId: "0001",
+						Name:      "Shankar",
+						LastName:  "Nakai",
+					},
+					"0002": {
+						AccountId: "0002",
+						Name:      "Jessica",
+						LastName:  "Lourenco",
+					}},
+			},
+
+			want: models.Account{
+				AccountId: id,
+				Name:      "Caio",
+				LastName:  "Henrique",
+			},
+			wantErr: nil,
+		},
+		"missing params": {
```

Comment:
You should test for missing name and another for missing last name amd one for both